manifests/charts/istio-cni/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,53 @@
+{{- if (.Values.networkPolicy).enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "name" . }}{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: {{ template "name" . }}-node
+    release: {{ .Release.Name }}
+    istio.io/rev: {{ .Values.revision | default "default" | quote }}
+    install.operator.istio.io/owning-resource: {{ .Values.ownerName | default "unknown" }}
+    operator.istio.io/component: "Cni"
+    app.kubernetes.io/name: {{ template "name" . }}
+    {{- include "istio.labels" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      k8s-app: {{ template "name" . }}-node
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  # Metrics endpoint for monitoring/prometheus
+  - from: []
+    ports:
+    - protocol: TCP
+      port: 15014
+  # Readiness probe endpoint
+  - from: []
+    ports:
+    - protocol: TCP
+      port: 8000
+  egress:
+  # Allow DNS resolution
+  - to: []
+    ports:
+    - protocol: UDP
+      port: 53
+    - protocol: TCP
+      port: 53
+  # Allow access to Kubernetes API server in kube-system namespace
```

Comment:
I am not confident this works in most environments. The api-server is not always in kube-system nor a pod. Nor on these ports. Unless there is documentation suggesting a supported portable way to allow access here this seems problematic

Reply:
One more option is...

to allow users to customize the policy in `values.yaml`
```
networkPolicy:
  # When enabled, a default NetworkPolicy for istio-cni will be created
  enabled: false
  # Override the default NetworkPolicy with custom rules
  # If specified, this will replace the default egress rules
  customEgress: []
```

And update the K8s NetworkPolicy template to use the customEgress rules (if specified)  
```
  egress:
  {{- if .Values.networkPolicy.customEgress }}
    {{- toYaml .Values.networkPolicy.customEgress | nindent 2 }}
  {{- else }}
  # Allow all egress traffic by default
  - {}
  {{- end }}  
```

Reply:
Fine with me @dgn. Makes sense to start small and expand it in future depending on the requirements.